lib/sprockets/server.rb

```diff
@@ -81,6 +81,14 @@ def call(env)
         logger.info "#{msg} 412 Precondition Failed (#{time_elapsed.call}ms)"
         precondition_failed_response
       end
+
+      if env['REQUEST_METHOD'] == 'HEAD'
+        response_status, response_headers, _ = rack_response
+        response_headers["Content-Length"] = "0" unless response_headers.nil?
+        [response_status, response_headers, []]
+      else
```

Comment:
Yeah, we should handle the HEAD requests inside of each response method not here as an special way.